src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,31 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
 
 public class CountryServiceImpl implements CountryService {
+    private CountryDao countryDao;
+
+    public CountryServiceImpl(CountryDao countryDao) {
+        this.countryDao = countryDao;
+    }
+
     @Override
     public Country add(Country country) {
-        return null;
+        if (country == null) {
+            throw new DataProcessingException("Cannot add null data");
+        }
+        countryDao.add(country);
+        return country;
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        if (id == null) {
+            throw new DataProcessingException("Cannot get by null id");
+        }
+        return countryDao.get(id).orElseThrow(DataProcessingException::new);
```

Comment:
I believe the custom exception is supposed to be thrown on Dao layer only. Here it would be better to throw NoSuchElementException with some message